src/peer_manager.rs

```diff
@@ -962,6 +965,10 @@ impl PeerManager {
         true
     }
 
+    pub fn mark_as_failed(&mut self, peer_id: &PeerId) -> bool {
```

Comment:
I'm not sure there's any point in this returning a `bool` since in the only place it's called we ignore the returned value.